docs/src/manual/models.md

```diff
@@ -62,50 +62,72 @@ functions.
 
 Consult the docstsrings of each attribute for information on what it represents.
 
-## Model attributes
+## ModelLike API
 
-The following attributes are available for models:
+Objects subtyping [`ModelLike`](ref) must implement:
+
+ * [`is_empty`](@ref)
+ * [`empty!`](@ref)
+
+The following attributes are available. Some are required by all
+implementations, and others are optional.
+
+**Required**
+
+ * [`ListOfConstraintAttributesSet`](@ref)
+ * [`ListOfConstraintIndices`](@ref)
+ * [`ListOfConstraintTypesPresent`](@ref)
+ * [`ListOfModelAttributesSet`](@ref)
+ * [`ListOfVariableAttributesSet`](@ref)
+ * [`ListOfVariableIndices`](@ref)
+ * [`NumberOfConstraints`](@ref)
+ * [`NumberOfVariables`](@ref)
+
+**Optional**
 
  * [`Name`](@ref)
  * [`ObjectiveFunction`](@ref)
  * [`ObjectiveFunctionType`](@ref)
  * [`ObjectiveSense`](@ref)
- * [`NumberOfVariables`](@ref)
- * [`ListOfVariableIndices`](@ref)
- * [`ListOfConstraintTypesPresent`](@ref)
- * [`NumberOfConstraints`](@ref)
- * [`ListOfConstraintIndices`](@ref)
- * [`ListOfOptimizerAttributesSet`](@ref)
- * [`ListOfModelAttributesSet`](@ref)
- * [`ListOfVariableAttributesSet`](@ref)
- * [`ListOfConstraintAttributesSet`](@ref)
 
-## Optimizer attributes
-
-The following attributes are available for optimizers:
-
- - [`SolverName`](@ref)
- - [`RawOptimizerAttribute`](@ref)
- - [`Silent`](@ref)
- - [`TimeLimitSec`](@ref)
- - [`NumberOfThreads`](@ref)
- - [`RawSolver`](@ref)
-
-In addition, optimizers should implement the following attributes to provide
-access to solutions:
-
- - [`TerminationStatus`](@ref)
- - [`TerminationStatusCode`](@ref)
- - [`PrimalStatus`](@ref)
- - [`DualStatus`](@ref)
- - [`ResultStatusCode`](@ref)
- - [`RawStatusString`](@ref)
- - [`ResultCount`](@ref)
- - [`ObjectiveValue`](@ref)
- - [`DualObjectiveValue`](@ref)
- - [`ObjectiveBound`](@ref)
- - [`RelativeGap`](@ref)
- - [`SolveTimeSec`](@ref)
- - [`SimplexIterations`](@ref)
- - [`BarrierIterations`](@ref)
- - [`NodeCount`](@ref)
+!!! note
+    To be useful, a [`ModelLike`](@ref) should, in most cases, also implement
+    [`copy_to`](@ref) or the incremental interface ([`add_variable`](@ref),
+    [`add_constraint`](@ref), etc.) See [Implementing a solver interface](@ref)
+    for more details.
+
+## AbstractOptimizer API
+
+There are two options for [`AbstractOptimizer`](@ref)s:
+
+ 1. Implement the required portion of the [ModelLike API](@ref), plus
```

Comment:
If you rephrase the above section as suggested, you can here say that an `AbstractOptimizer` should either 1) be writable (e.g., implement `copy_to` or incremental interface) + implement `optimize!` or 2) implement `copy_to_and_optimize!`